client/stripe-link/index.js

```diff
@@ -52,39 +52,75 @@ const enableStripeLinkPaymentMethod = ( options ) => {
 			  };
 
 		if ( options.complete_shipping() ) {
+			const shippingNames = shippingAddress.name.split( / (.*)/s, 2 );
+			shippingAddress.address.last_name = shippingNames[ 1 ];
+			shippingAddress.address.first_name = shippingNames[ 0 ];
+
 			fillWith( shippingAddress, options.shipping_fields.line1, 'line1' );
 			fillWith( shippingAddress, options.shipping_fields.line2, 'line2' );
 			fillWith( shippingAddress, options.shipping_fields.city, 'city' );
-			fillWith( shippingAddress, options.shipping_fields.state, 'state' );
 			fillWith(
 				shippingAddress,
-				options.shipping_fields.postal_code,
-				'postal_code'
+				options.shipping_fields.country,
+				'country'
 			);
 			fillWith(
 				shippingAddress,
-				options.shipping_fields.country,
-				'country'
+				options.shipping_fields.first_name,
+				'first_name'
+			);
+			fillWith(
+				shippingAddress,
+				options.shipping_fields.last_name,
+				'last_name'
+			);
+			jQuery(
+				'#billing_country, #billing_state, #shipping_country, #shipping_state'
+			).trigger( 'change' );
+			fillWith( shippingAddress, options.shipping_fields.state, 'state' );
+			fillWith(
+				shippingAddress,
+				options.shipping_fields.postal_code,
+				'postal_code'
 			);
```

Comment:
Without knowledge of codebase I don't understand 
- what this block does
- why 
- how is that related to the changes of this PR

Same comment about block below.

Reply:
this block addresses the state and postal code field. I did a small refactor, and replaced the `jQuery( 'select' ).trigger( 'change' );` that has a generic selector with a more targeted one.